onnxruntime/python/tools/transformers/optimizer.py

```diff
@@ -147,6 +147,7 @@ def optimize_by_onnxruntime(
             gpu_ep.append("ROCMExecutionProvider")
         else:
             gpu_ep.append("CUDAExecutionProvider")
+            gpu_ep.append("DmlExecutionProvider")
```

Comment:
I suggest to add a function parameter 
```
provider=None
```
If it is provided, use the specified execution provider. Otherwise, just pick the first provider from onnxruntime.get_available_providers() as default.